frontends/web/src/routes/account/summary/chart.tsx

```diff
@@ -14,28 +14,124 @@
  * limitations under the License.
  */
 
-import { createChart, IChartApi, LineData } from 'lightweight-charts';
+import { createChart, IChartApi, LineData, LineStyle, ISeriesApi, UTCTimestamp } from 'lightweight-charts';
 import { Component, createRef, h, RenderableProps } from 'preact';
 import { translate, TranslateProps } from '../../../decorators/translate';
+import { formatAsCurrency, getActiveFiat } from '../../../components/rates/rates';
+import styles from './chart.css';
 
 export type ChartData = LineData[];
 
 interface ChartProps {
     // If undefined, data is missing.
-    data?: ChartData;
+    dataDaily?: ChartData;
+    dataHourly?: ChartData;
 }
 
 interface State {
+    display: 'week' | 'month' | 'year' | 'all';
+    source: 'daily' | 'hourly';
+    currentTotal?: number;
+    difference?: number;
+    diffSince?: string;
 }
 
 type Props = ChartProps & TranslateProps;
 
 class Chart extends Component<Props, State> {
     private ref = createRef();
     private chart?: IChartApi;
+    private lineSeries?: ISeriesApi<'Area'>;
     private resizeTimerID?: any;
     private height: number = 300;
 
+    public readonly state: State = {
+        display: 'all',
+        source: 'daily',
+        currentTotal: this.props.dataDaily && this.props.dataDaily[this.props.dataDaily.length - 1].value,
+    };
+
+    public componentDidMount() {
+        if (this.ref.current) {
+            this.chart = createChart(this.ref.current, {
+                width: this.ref.current.offsetWidth,
+                height: this.height,
+                handleScroll: false,
+                handleScale: false,
+                crosshair: {
+                    vertLine: {
+                        visible: false,
+                        labelVisible: true, // TODO: only enabled for debug
+                    },
+                    horzLine: {
+                        visible: false,
+                        labelVisible: true, // TODO: only enabled for debug
+                    },
+                    mode: 1,
+                },
+                grid: {
+                    vertLines: {
+                        visible: false,
+                    },
+                    horzLines: {
+                        color: '#dedede',
+                        style: LineStyle.Solid,
+                        visible: true,
+                    },
+                },
+                layout: {
+                    backgroundColor: '#F5F5F5',
+                    fontSize: 11,
+                    fontFamily: '-apple-system, BlinkMacSystemFont, "Segoe UI", "Ubuntu", "Roboto", "Oxygen", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif',
+                    textColor: '#1D1D1B',
+                },
+                leftPriceScale: {
+                    // mode: PriceScaleMode.Logarithmic, // TODO: future feature
```

Comment:
delete? can always add if needed, but this looks like a candidate for a TODO that might never be addressed ;)